src/ape/api/projects.py

```diff
@@ -0,0 +1,259 @@
+import json
+from pathlib import Path
+from typing import TYPE_CHECKING, Collection, Dict, List, Optional
+
+from ethpm_types import Checksum, ContractType, PackageManifest, Source
+from ethpm_types.manifest import PackageName
+from ethpm_types.utils import compute_checksum
+from packaging import version as version_util
+from pydantic import ValidationError
+
+from ape.logging import logger
+from ape.utils import (
+    abstractdataclass,
+    abstractmethod,
+    get_all_files_in_directory,
+    get_relative_path,
+)
+
+if TYPE_CHECKING:
+    from ape.managers.compilers import CompilerManager
+    from ape.managers.project import ProjectManager
+
+
+@abstractdataclass
+class ProjectAPI:
+    """
+    An abstract base-class for working with projects.
+    This class can also be extended to a plugin for supporting non-ape projects.
+    """
+
+    compilers: "CompilerManager"
```

Comment:
There is no base API model yet and `injected_before_use()` does not seem to work with `abstractdataclass`.
I am trying to do this but struggling for reasons like this.

Since all the API classes work with way right now, can we do this at the same time as those, as more research is required there.

Reply:
for sure that works